arm-sql/2014-04-01/swagger/dataMasking.json

```diff
@@ -0,0 +1,460 @@
+                        "name": "parameters",
+                        "in": "body",
+                        "required": true,
+                        "schema": {
+                            "$ref": "#/definitions/DataMaskingRule"
+                        },
+                        "description": "The required parameters for createing or updating a data masking rule."
+                    }
+                ],
+                "responses": {
+                    "200": {
+                        "description": "",
+                        "schema": {
+                            "$ref": "#/definitions/DataMaskingRule"
+                        }
+                    },
+                    "201": {
+                        "description": "",
+                        "schema": {
+                            "$ref": "#/definitions/DataMaskingRule"
+                        }
+                    }
+                }
+            },
+            "delete": {
+                "tags": [
+                    "DataMasking"
+                ],
+                "operationId": "DataMasking_DeleteRule",
+                "description": "Deletes an Azure SQL database data masking rule.",
+                "parameters": [
+                    {
+                        "$ref": "#/parameters/ApiVersionParameter"
+                    },
+                    {
+                        "$ref": "#/parameters/SubscriptionIdParameter"
+                    },
+                    {
+                        "$ref": "#/parameters/ResourceGroupParameter"
+                    },
+                    {
+                        "$ref": "#/parameters/ServerNameParameter"
+                    },
+                    {
+                        "name": "databaseName",
+                        "in": "path",
+                        "required": true,
+                        "type": "string",
+                        "description": "The name of the Azure SQL Database for which the data masking rule applies."
+                    },
+                    {
+                        "name": "dataMaskingRule",
+                        "in": "path",
+                        "required": true,
+                        "type": "string",
+                        "description": "The name of the Azure SQL Database  data masking rule."
+                    }
+                ],
+                "responses": {
+                    "204": {
+                        "description": ""
+                    },
+                    "200": {
+                        "description": ""
+                    }
+                }
+            }
+        },
+        "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Sql/servers/{serverName}/databases/{databaseName}/dataMaskingPolicies/Default/Rules": {
+            "get": {
+                "tags": [
+                    "DataMasking"
+                ],
+                "operationId": "DataMasking_ListRules",
```

Comment:
The only option is to list the rules in a database. Is the ByDatabase still recommended (ie: we don't do Database_ListByServer)?

---
In reply to: [80148856](https://github.com/Azure/azure-rest-api-specs/pull/555#discussion_r80148856) [](ancestors = 80148856)